File: core/src/main/java/me/mintytc/azapi/core/interfaces/Transformable.java
package me.mintytc.azapi.core.interfaces;

import java.util.function.Function;

/**
 * @since 1.0.0-R0.1-BASE
 *
 */
public interface Transformable<T> {
    /**
     * Transforms this object into another of the same type.
     *
     * @param transformer function describing how to transform
     *
     * @return transformed object
     */
    T transform(Function<T, T> transformer);
}

File: src/main/java/me/mintytc/azapi/interfaces/Transformable.kt
package me.mintytc.azapi.interfaces

import java.util.function.Function

/**
 * @since 1.0.0-R0.1
 */
interface Transformable<T> {
	/**
	 * Transforms this object into another of the same type.
	 *
	 * @param transformer function describing how to transform
	 *
	 * @return transformed object
	 */
	fun transform(transformer: Function<T?, T?>?): T?
}